src/compute/src/memory/config.rs

```diff
@@ -107,24 +114,60 @@ pub fn storage_memory_config(
     } else {
         (STORAGE_MEMORY_PROPORTION + COMPACTOR_MEMORY_PROPORTION, 0.0)
     };
-    let mut default_block_cache_capacity_mb = ((non_reserved_memory_bytes as f64
-        * storage_memory_proportion
-        * STORAGE_BLOCK_CACHE_MEMORY_PROPORTION)
-        .ceil() as usize)
-        >> 20;
-    let default_meta_cache_capacity_mb = ((non_reserved_memory_bytes as f64
-        * storage_memory_proportion
-        * STORAGE_META_CACHE_MEMORY_PROPORTION)
-        .ceil() as usize)
-        >> 20;
-    let meta_cache_capacity_mb = storage_config.cache.meta_cache_capacity_mb.unwrap_or(
+
+    let storage_memory_bytes = non_reserved_memory_bytes as f64 * storage_memory_proportion;
+
+    // Only if the all cache capacities are specified and their sum doesn't exceed the max allowed storage_memory_bytes, will we use them.
+    // Other invalid combination of the cache capacities config will be ignored.
+    let (
+        config_block_cache_capacity_mb,
+        config_meta_cache_capacity_mb,
+        config_shared_buffer_capacity_mb,
+    ) = match (
+        storage_config.cache.block_cache_capacity_mb,
+        storage_config.cache.meta_cache_capacity_mb,
+        storage_config.shared_buffer_capacity_mb,
+    ) {
+        (
+            Some(block_cache_capacity_mb),
+            Some(meta_cache_capacity_mb),
+            Some(shared_buffer_capacity_mb),
+        ) => {
+            let config_storage_memory_bytes =
+                (block_cache_capacity_mb + meta_cache_capacity_mb + shared_buffer_capacity_mb)
+                    << 20;
+            if config_storage_memory_bytes as f64 > storage_memory_bytes {
```

Comment:
This is a good restriction, but directly ignoring incoming misconfigurations may cause users to overlook errors.

In short: Do we need to use a panic instead of a warn log to stop the system from running when an exception occurs in the configuration, in order to get the user's attention?

Reply:
The main motivation of this PR is to avoid panic and OOM when user scale-in compute node with the cache configs specified explicitly in the config.

Reply:
Both are fine with me.